modules/core/src/main/scala/com.snowplowanalytics.snowplow/scalatracker/Tracker.scala

```diff
@@ -36,7 +36,7 @@ import scala.concurrent.duration._
 final case class Tracker[F[_]: Monad: Clock: UUIDProvider](emitters: NonEmptyList[Emitter[F]],
```

Comment:
1. I don't think it's a proper case class, e.g. we cannot provide meaningful `toString` and `equals`.
2. In more tagless-final-ish approach we have type class constraints on function not on data types

Both are minor though and I think addressing 2nd point might require a big API change.

Reply:
I changed to a regular class, not case class.

I chose to keep the class constraints on the class, rather than the functions.  The other way around would require the developer to have a UUIDProvider and Clock in scope every time they call `tracker.trackSomething(....)`.  And for the id-emitters I am not fully satisfied with how we provide a UUIDProvider[Id] and a Clock[Id] - so I would prefer it if these are required only for the tracker constructors, rather than throughout the developer's entire codebase.

Good suggestions though!!